pkg/label/label.go

```diff
@@ -13,3 +13,29 @@ const (
 	// ManagedBy is set for Kubernetes resources managed by the operator.
 	ManagedBy = "giantswarm.io/managed-by"
 )
+
+// ProcessLabels ensures the chart-operator.giantswarm.io/version label is
+// present and the app-operator.giantswarm.io/version label is removed. It
+// also ensures the giantswarm.io/managed-by label is accurate.
```

Comment:
This comment is pretty nice, and it shows that the method does too much IMO. The primitives here are defaulting and validation. The code below is simple in this very use case but we should try to separate and isolate implementations so that they are maintainable and testable. When I test two conceptually different functionalities and change one implementation, breaking two tests shows that the design is wrong. A little bit of docs and implementations can be found in various places. 

* https://github.com/giantswarm/giantswarm/blob/master/sigs/operator/command_line_tools.md
* https://github.com/giantswarm/api/tree/master/service/cluster/creator/defaulting
* https://github.com/giantswarm/api/tree/master/service/cluster/creator/validation
* https://github.com/giantswarm/api/tree/master/service/cluster/updater/defaulting
* https://github.com/giantswarm/api/tree/master/service/cluster/updater/validation

Reply:
The current comment is very detailed and only makes for the chart resource. I agree the logic should be generic and basing it on our existing implementations is a good approach.

Reply:
I agree I'll make it smaller then.